src/utils/transaction.ts

```diff
@@ -934,6 +1137,210 @@ export const buildCrossChainAction = async (
     }
   }
 
+  if (
+    transactionBlock.type === TRANSACTION_BLOCK_TYPE.HONEY_SWAP_LP &&
+    !!transactionBlock?.values?.fromChainId &&
+    !!transactionBlock?.values?.toToken1 &&
+    !!transactionBlock?.values?.toToken2 &&
+    !!transactionBlock?.values?.fromAssetAddress &&
+    !!transactionBlock?.values?.fromAssetDecimals &&
+    !!transactionBlock?.values?.fromAssetSymbol &&
+    !!transactionBlock?.values?.amount &&
+    !!transactionBlock?.values?.routeToUSDC &&
+    !!transactionBlock?.values?.offer1 &&
+    !!transactionBlock?.values?.offer2 &&
+    !!transactionBlock?.values?.tokenOneAmount &&
+    !!transactionBlock?.values?.tokenTwoAmount
+  ) {
+    try {
+      const {
+        values: {
+          fromChainId,
+          fromAssetAddress,
+          fromAssetDecimals,
+          fromAssetSymbol,
+          fromAssetIconUrl,
+          amount,
+          accountType,
+          routeToUSDC,
+          receiverAddress,
+          offer1,
+          offer2,
+          tokenOneAmount,
+          tokenTwoAmount,
+          toToken1,
+          toToken2,
+        },
+      } = transactionBlock;
+
+      const USDC_GNOSIS = '0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83';
+
+      if (fromChainId !== CHAIN_ID.XDAI) {
+        try {
+          let destinationTxns: ICrossChainActionTransaction[] = [];
+          let transactions: ICrossChainActionTransaction[] = [];
+
+          console.log("ETHERSPOT:::8")
+
+          const [firstStep] = routeToUSDC.steps;
+          const bridgeServiceDetails = bridgeServiceIdToDetails[firstStep?.toolDetails?.key ?? ''];
+
+          const { items: advancedRouteSteps } = await sdk.getStepTransaction({ route: routeToUSDC });
+
+          transactions = advancedRouteSteps.map(({ to, value, data, chainId }) => ({
+            to: to as string,
+            value,
+            data,
+            chainId: chainId ?? fromChainId,
+            createTimestamp,
+            status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+          }));
+
+          if (
+            ethers.utils.isAddress(routeToUSDC.fromToken.address) &&
```

Comment:
Again sorry, just some null checks
```suggestion
            ethers.utils.isAddress(routeToUSDC?.fromToken?.address) &&
```

Reply:
Null check isn't required here we're checking it at the top of the function.